### DISCUSSION TITLE
A Go-wrapper around the DBus API

### DISCUSSION THREAD

**Discussion by Jacalz** [flatpak/xdg-desktop-portal#1308] [Ideas]
Hello everyone 👋 

I have created the [portal](https://github.com/rymdport/portal ) package to allow an easy, typed Go library for interacting with the XDG Desktop Portal protocol from Go without linking to libportal through CGo. It is intended to be useable by Fyne, Gio, Wails or any other graphical app toolkit that may wish to use it.

I hope people find it useful. I've tried to make the API as close to the DBus API as possible and also focused on ease of use and clean code.

**Comment by GeorgesStavracas** [comment]
Nice! Very exciting to see this kind of integration work happening

**Comment by hfiguiere** [comment]
Maybe we should add a section on the website about client implementations.

**Reply by Jacalz** [reply]
I think that would be a great idea. The only other one that I personally know of is https://github.com/bilelmoussaoui/ashpd.
